src/Type/Php/DsMapDynamicReturnTypeExtension.php

```diff
@@ -35,11 +35,11 @@ public function getTypeFromMethodCall(MethodReflection $methodReflection, Method
 			)->getReturnType();
 		}
 
-		if (! $returnType instanceof UnionType) {
-			return $returnType;
+		if ($returnType instanceof UnionType) {
+			return new UnionType(array_slice($returnType->getTypes(), 0, -1));
```

Comment:
This doesn't seem right at all.

1) You can't assume what will be the last element in an array in a UnionType.
2) UnionType of a single element is invalid.

Instead, you should get `->getTypes()`, filter the unwanted type (I guess `NullType`?) and then call `TypeCombinator::union(...$types)` on the rest.

Reply:
With the new commit I filter out TDefault (I assume if present it means no value was provided)